src/crypto/public_key/elliptic/ecdh.js

```diff
@@ -84,20 +87,34 @@ async function kdf(hash_algo, X, length, param, stripLeading=false, stripTrailin
  * @async
  */
 async function genPublicEphemeralKey(curve, Q) {
-  if (curve.name === 'curve25519') {
-    const { secretKey: d } = nacl.box.keyPair();
-    const { secretKey, sharedKey } = await genPrivateEphemeralKey(curve, Q, d);
-    let { publicKey } = nacl.box.keyPair.fromSecretKey(secretKey);
-    publicKey = util.concatUint8Array([new Uint8Array([0x40]), publicKey]);
-    return { publicKey, sharedKey }; // Note: sharedKey is little-endian here, unlike below
+  switch (curve.name) {
+    case ('curve25519'): {
+      const { secretKey: d } = nacl.box.keyPair();
+      const { secretKey, sharedKey } = await genPrivateEphemeralKey(curve, Q, d);
+      let { publicKey } = nacl.box.keyPair.fromSecretKey(secretKey);
+      publicKey = util.concatUint8Array([new Uint8Array([0x40]), publicKey]);
+      return { publicKey, sharedKey }; // Note: sharedKey is little-endian here, unlike below
+    }
+    case ('p256'):
+    case ('p384'):
+    case ('p521'): {
+      let result;
+      if (curve.web && util.getWebCrypto()) {
+        try {
+          result = await webPublicEphemeralKey(curve, Q);
+        } catch (err) {
```

Comment:
Could you add `util.print_debug_error(err);` in this catch (and the other one) to help with debugging? (You can turn on printing those errors by adding `?debug=1` to the browsertests URL.)